src/FluentMigrator.Runner.Oracle/Generators/Oracle/OracleQuoterBase.cs

```diff
@@ -15,13 +15,40 @@
 #endregion
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
 
 using FluentMigrator.Runner.Generators.Generic;
 
 namespace FluentMigrator.Runner.Generators.Oracle
 {
     public class OracleQuoterBase : GenericQuoter
     {
+        private const int MaxStringLength = 3900;
+
+        private static IEnumerable<string> SplitBy(string str, int chunkLength)
```

Comment:
Is there no .NET library function that does this?

Reply:
I think we should use a version of string.Substring that works for Unicode.  See my research here: https://stackoverflow.com/a/61125710/1040437 and see: https://stackoverflow.com/a/8959444/1040437 (I've edited this answer so you might need to wait 24 hours for moderator approval to fix a minor nit with the answer).

Reply:
I didn't find any, but you seem to have found one, though, related to another review you made, I think we must split the string so that each chunk does not exceed 4000 bytes. Do you think it's possible to achieve this with `SubstringByTextElements` ?

Reply:
Ah!  I see the issue now.  I was reading MaxStringLength as characters, forgetting that it's a byte length limitation.  Shoot!  Good, good point.

I think it is possible to achieve this with SubstrinByTextElements - we just need a rolling accumulator variable that tracks how many bytes we've been passed from `StringInfo.ParseCombinigCharacters`.  Do you follow my explanation?

Reply:
Hello, I've been able to test this pull request on a Oracle 12.1 server, and fixed a few things : 
- MaxStringLength is now 4000
- As quotes are doubled during the escape sequence, we have to take this into account
- Oracle is OK with multibyte strings, so we don't have to worry about couting bytes or not (for CLOB and NCLOB ...)

Also, I've implemented the base code for EndToEnd integration tests on OracleManaged and a test for this pull request.

I'll tell you when it's ready, because I have to update the other unit tests as the test string changed to include Unicode chars

Reply:
Thanks, @PhenX - I'll try to merge it into 3.2.7 as opposed to 3.3.0, since its likely your tests won't impact my other changes I'm working on.